### PR TITLE
Add Armorable.is_armor(text) for detecting ASCII-armored data.

### PR DESCRIPTION
**Pull request by J08nY**
Adds a method to ease recognizing ASCII-armored data without fully parsing it, also adds test to check it works as intended.

### CODE REVIEW

**Comment by Commod0re** [comment]
I think I like where this is going. Is there a particular problem this solves, though?

**Comment by J08nY** [comment]
The currently best way of identifying whether some `str/bytes/bytearray` is ASCII-armored is (AFAIK) via:
```python
armored = True
try:
    Armorable.unarmor(text)
except ValueError:
    armored = False
```
Which seems unpythonic to me.

For example use, I'm currently implementing a RFC3156(PGP/MIME) enabled mailing list plugin for GNU Mailman. I would also like to have inline PGP support there. So tools for recognizing an armored part/s in some text, would be nice.

**Comment by Commod0re** [comment]
I see, thank you for clarifying. That makes perfect sense to me :)

File: pgpy/types.py [hunk 1]
```diff
@@ -53,6 +53,28 @@ class Armorable(six.with_metaclass(abc.ABCMeta)):
                     '={crc}\n' \
                     '-----END PGP {block_type}-----\n'
 
+    # the re.VERBOSE flag allows for:
+    #  - whitespace is ignored except when in a character class or escaped
+    #  - anything after a '#' that is not escaped or in a character class is ignored, allowing for comments
+    __armor_regex__ = re.compile(r"""# This capture group is optional because it will only be present in signed cleartext messages
```
Comment (Commod0re) [inline pgpy/types.py]: I like that you moved this to the class level. I think this should just be named `__armor_regex` so the name gets mangled to hide it from subclasses and not confused as a magic member instead

Comment (J08nY) [inline pgpy/types.py]: Yeah, definitely. Will change.

**Pull request merged**